src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_Utilities.cs

```diff
@@ -99,8 +98,10 @@ JToken ParameterFunction(ITextSnapshot snapshot)
         ///         ]
         ///     }
         /// ]
-        public static DocumentSymbolDataModel CreateDocumentSymbolDataModel(LspDocumentSymbol[] documentSymbols, ITextSnapshot originalSnapshot)
+        public static ImmutableArray<DocumentSymbolData> CreateDocumentSymbolData(JToken token, ITextSnapshot textSnapshot)
         {
+            var documentSymbols = token.ToObject<RoslynDocumentSymbol[]>() ?? Array.Empty<RoslynDocumentSymbol>();
```

Comment:
if this is null can we just early return? It doesn't look like anything will be added to the results and we can avoid getting an ArrayBuilder etc

Reply:
so, i don't want to early return as i want a null-result to clear out everything in the outline window.  Bailing early woudl cause us to keep stale things aorund.  i will doc.